.appveyor.yml

```diff
@@ -26,10 +31,22 @@ install:
   - cinst swig
   - cinst nsis
   - SET PATH=%PYTHON%;%PYTHON%\Scripts;%PATH%
+  - set PATH=C:\MinGW\bin;%PATH%
   - python --version
   # Upgrade to the latest pip.
   - '%CMD_IN_ENV% python -m pip install -U pip setuptools wheel tox'
   - powershell .\\tasks\\missing-headers.ps1
+  - md lib
+  - cd lib
+  - appveyor DownloadFile "https://raw.githubusercontent.com/swift-nav/libsettings/56ac4611e9fb148d7beaf469384247872f056f6d/python/libsettings-0.1.0.tar.gz"
+  - md libsettings
+  - tar -xzf libsettings-0.1.0.tar.gz -C libsettings --strip-components=1
+  - cd libsettings
+  # Hack to get rid of format specifier warnings emerging from the cythonized file
+  - C:\cygwin64\\bin\sed.exe -i 's/#define CYTHON_FORMAT_SSIZE_T "z"/#define CYTHON_FORMAT_SSIZE_T "I"/g' python/libsettings.c
+  - C:\cygwin64\\bin\sed.exe -i "s/zd /Id /g" python/libsettings.c
+  - python setup.py build_ext --inplace --compiler=mingw32
+  - cd ../..
```

Comment:
What does this mean from a "pip install piksi_tools" perspective on a windows machine?  

Right now you can pip install the command line tools anywhere and they just work.  Custmoers use them to programmatically configure devices.  Does libsettings have its own command line executable that people should use instead?  

It seems to me that installing piksi_tools on windos is going  to get much harder with libsettings.  Is it possible?  If so, what is the procedure?  Customer's likely won't have mingw, cygwin, or sed; in that case, could they install  piksi_tools and get access to a settings command line client?

Reply:
No command line executable. I need to investigate what actually happens with `pip install piksi_tools`.

Few comments on the spot:

`cygwin` is only needed for `sed`. `sed` is needed only for getting rid of format specifier compatibility warnings from cython source error messages so not a hard requirement. I first tried to do the replacement with powershell tools but windows cmd line is very picky with the quotes and escape chars. I managed to get it working locally but not on appveyor environment. *enter sarcastic comments about unix shell vs win cmd*

`mingw` is a must if we want to have `c99` in libsettings, for a moment I already started writing `c89`/`c90` compatible code but... it just feels so dumb. In case of Python3 windows might be using newer compiler that actually supports `c99`, not sure about this.

Reply:
I did some investigations. It is unclear for me if `python -m piksi_tools.console.console` is expected to work on current `master` after running `pip install piksi_tools` on Windows. See slack conversation I started https://snav.slack.com/archives/C0F4WHXK5/p1546249486001200

If use case is not expected to work then this issue is also irrelevant regarding `settings_view.py`. Although this means that settings.py tool can't be upgraded to use libsettings at this stage.

Reply:
I think the closest thing what you mean by libsettings command line client is `settings_client.py` in https://github.com/swift-nav/libsettings/pull/3

Nonetheless it requires building `libsettings` first. Adding win build support is possible.

I suggest we leave `piksi_tools/settings.py` tool as is for now and in parallel work to get similar functionality and ease of use with `libsettings/settings_client.py` in subsequent PRs.

Reply:
I think that is fine, but we'd need to eventually get libsettings able to install via pip for any"pip install" based deployment to work, right?  This would mean deploying to PyPI.

Reply:
I think this captures the issue? https://swift-nav.atlassian.net/browse/ESD-988

Reply:
Yes and yes (Y)